plugins/inputs/prometheus_scraper/metric_prometheus.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package prometheus_scraper
+
+import (
+	"github.com/prometheus/prometheus/model/textparse"
+	"github.com/prometheus/prometheus/model/value"
+	"math"
+)
+
+type PrometheusMetricBatch []*PrometheusMetric
+
+type PrometheusMetric struct {
+	tags        map[string]string
+	metricName  string
+	metricValue float64
+	metricType  string
+	timeInMS    int64 // Unix time in milli-seconds
+}
+
+func (pm *PrometheusMetric) isValueValid() bool {
+	//treat NaN and +/-Inf values as invalid as emf log doesn't support them
+	return !value.IsStaleNaN(pm.metricValue) && !math.IsNaN(pm.metricValue) && !math.IsInf(pm.metricValue, 0)
+}
+
+func (pm *PrometheusMetric) isCounter() bool {
+	return pm.metricType == string(textparse.MetricTypeCounter)
+}
+
+func (pm *PrometheusMetric) isGauge() bool {
+	return pm.metricType == string(textparse.MetricTypeGauge)
+}
+
+func (pm *PrometheusMetric) isHistogram() bool {
+	return pm.metricType == string(textparse.MetricTypeHistogram)
+}
+
+func (pm *PrometheusMetric) isSummary() bool {
+	return pm.metricType == string(textparse.MetricTypeSummary)
+}
```

Comment:
Where are these functions used?

Reply:
It is used in the [Filter](https://github.com/khanhntd/amazon-cloudwatch-agent/blob/master/plugins/inputs/prometheus_scraper/metrics_filter.go#L21-L48) which drops the unsupported metrics and also used in [calculation](https://github.com/khanhntd/amazon-cloudwatch-agent/blob/master/plugins/inputs/prometheus_scraper/calculator.go#L24-L53)

**For more context in the delta calculation**: The reason why we would perform some calculation (e.g Delta with Counter Prometheus Type) since there is no support for Counter Type in CWMetrics. The only way we can do is calculate the difference and send it over to CW Backend.